actioncable/lib/action_cable/server/connection.rb

```diff
@@ -0,0 +1,198 @@
+# frozen_string_literal: true
+
+require "action_dispatch"
+
+module ActionCable
+  module Server
+    # This class encapsulates all the low-level logic of working with the underlying WebSocket conenctions
+    # and delegate all the business-logic to the user-level connection object (e.g., ApplicationCable::Connection).
+    # This connection object is also responsible for handling encoding and decoding of messages, so the user-level
+    # connection object shouldn't know about such details.
+    class Connection
+      attr_reader :server, :env, :protocol, :logger, :app_conn
```

Comment:
I wonder if we can avoid abbreviations/improve the naming of `:app_conn`.

Reply:
Renamed to "connection" (since it's an instance of ActionCable::Connection::Base)